diffsims/generators/structure_library_generator.py

```diff
@@ -42,68 +28,29 @@ class StructureLibraryGenerator:
     phases : list
         Array of three-component phase descriptions, where the phase
         description is [<phase name> : string, <structure> :
-        diffpy.structure.Structure, <crystal system> : string], and crystal
-        system is one of 'cubic', 'hexagonal', 'orthorhombic', 'tetragonal',
-        'trigonal', 'monoclinic'.
+        diffpy.structure.Structure, <rotation_list> : list]
 
     Attributes
     ----------
     phase_names : list of string
         List of phase names.
     structures : list of diffpy.structure.Structure
         List of structures.
-    systems : list of string
-        List of crystal systems.
+    orientation : list of lists
 
-    Examples
-    --------
-    >>> gen = StructureLibraryGenerator([
-    ...     ('ZB', structure_zb, 'cubic'),
-    ...     ('WZ', structure_wz, 'hexagonal')])
     """
 
     def __init__(self, phases):
         self.phase_names = [phase[0] for phase in phases]
         self.structures = [phase[1] for phase in phases]
-        self.systems = [phase[2] for phase in phases]
+        self.orientations = [phase[2] for phase in phases]
 
-    def get_orientations_from_list(self, orientations):
-        """Create a structure library from a list of rotations.
-
-        Parameters
-        ----------
-        orientations : list
-            A list over identifiers of lists of euler angles (as tuples) in the rzxz
-            convention and in degrees.
+    def get_library(self):
+        """Create a structure library
 
         Returns
         -------
         structure_library : StructureLibrary
             Structure library for the given phase names, structures and orientations.
         """
-        return StructureLibrary(self.phase_names, self.structures, orientations)
-
-    def get_orientations_from_stereographic_triangle(self, inplane_rotations, resolution):
```

Comment:
Why is this function getting fully removed? My impression is that this approach is the one used in the Meng & Zuo paper about improving automated indexation and in Rauch papers - so I doubt it's wrong in principle and in the short term, I'd have a preference for supporting it as a "community standard", but only if it's not a massive time sink.

Reply:
I missed this at first pass. I'm happy to keep it, will undo the deletion and make corrections in a different PR